PsimagLite/examples/testArnoldiSaI.cpp

```diff
@@ -0,0 +1,187 @@
+	std::vector<ComplexType> eigenvalues(n);
+	PsimagLite::Matrix<ComplexOrRealType> vl_unused(n, n);
+	PsimagLite::Matrix<ComplexOrRealType> right_eigenvectors(n, n);
+	geev('N', 'V', m_copy, eigenvalues, vl_unused, right_eigenvectors);
+
+	// Matrix eigenvalues are these
+	std::cout << "Actual eigs\n";
+	std::cout << eigenvalues << "\n\n";
+
+	unsigned int the_one = findTheOne(eigenvalues);
+	assert(the_one < eigenvalues.size());
+	std::cout << "Lowest eig of L = " << the_one << " " << eigenvalues[the_one] << "\n";
+	printEigenvector(std::cout, right_eigenvectors, the_one);
+
+	// Now we do Arnoldi Shift-and-Invert
+	double sigma = 0.1; // small value here
+
+	// shift first
+	m_copy = m;
+	for (int i = 0; i < n; ++i) {
+		m_copy(i, i) -= sigma;
+	}
+
+	// invert now
+	inverse(m_copy);
+
+	// Eigenvalues of the inverse
+	PsimagLite::Matrix<ComplexOrRealType> matrix_inverse(m_copy);
+	std::vector<ComplexType> inverse_eigs(n);
+	geev('N', 'V', m_copy, inverse_eigs, vl_unused, right_eigenvectors);
+	std::cout << "Inverse eigs\n";
+	std::cout << inverse_eigs << "\n\n";
+
+	// Predict for comparison the eigenvalues of the sai
+	std::vector<ComplexType> predicted_eigs(n);
+	for (int i = 0; i < n; ++i) {
+		predicted_eigs[i] = 1.0 / (eigenvalues[i] - sigma);
+	}
+
+	std::cout << "Predicted eigs\n";
+	std::cout << predicted_eigs << "\n\n";
+
+	// Finally do Arnoldi
+	/* We convert the dense matrix into sparse
+	 * Obviously this is unrealistic, we would
+	 * start with a CRS matrix from the go.
+	 * But here for simplicify of filling I
+	 * chose to fill first a dense matrix and then convert,
+	 * and also for ease of debugging */
+	PsimagLite::CrsMatrix<ComplexOrRealType> msparse(matrix_inverse);
+
+	/* These are the parameters that control PsimagLite's Lanczos solver */
+	SolverParametersType params;
+
+	/* We store the resulting Lanczos vectors in RAM */
+	params.lotaMemory = true;
+	params.tolerance = 1e-8;
+
+	/* We create the solver object */
+	PsimagLite::ArnoldiIteration<PsimagLite::CrsMatrix<ComplexOrRealType>> arnoldi(params);
+
+	/* This is the initial vector for Lanczos;
+	 * we here set it to random */
+	VectorType initial(n);
+	PsimagLite::fillRandom(initial);
+
+	std::vector<VectorType> q;
+	PsimagLite::Matrix<ComplexOrRealType> h;
+	SizeType n_arnoldi = 50;
+	int error = arnoldi.decompose(q, h, msparse, initial, n_arnoldi);
+	std::cerr << "Arnolid return code = " << error << "\n";
+
+	// Now we diag h
+	std::vector<ComplexOrRealType> eigs_of_h;
+	PsimagLite::Matrix<ComplexOrRealType> h_right_eigenvectors;
+	arnoldi.diagHessenberg(eigs_of_h, h_right_eigenvectors, h);
+	std::cout << "Eigs of h\n";
+	std::cout << eigs_of_h << "\n\n";
+
+	std::vector<ComplexOrRealType> l_eigenvector_estimate(n);
+	arnoldi.getEigenvector(l_eigenvector_estimate, h_right_eigenvectors, q, 0);
+	// makeItReal(l_eigenvector_estimate);
+	std::cout << "\nEstimated eigenvector \n";
+	std::cout << l_eigenvector_estimate << "\n";
```

Comment:
It would be preferable to go ahead and use Catch2 in this test and actually assert what these eigenvalues and eigenvectors are expected to be instead of just printing to the screen.
Any good reason to defer that work?